lib/xenops_server.ml

```diff
@@ -2319,27 +2370,19 @@ module VM = struct
     let memory_limit = List.assoc "memory_limit" cookies |> Int64.of_string in
     Debug.with_thread_associated dbg
       (fun () ->
-         let is_localhost, id = Debug.with_thread_associated dbg
+         let id, final_id = Debug.with_thread_associated dbg
              debug "VM.receive_memory";
-           let remote_instance = List.assoc "instance_id" cookies in
-           let is_localhost = instance_id = remote_instance in
+           let final_id = List.assoc "final_id" cookies in
            (* The URI is /service/xenops/memory/id *)
            let bits = Stdext.Xstringext.String.split '/' (Uri.path uri) in
            let id = bits |> List.rev |> List.hd in
-           debug "VM.receive_memory id = %s is_localhost = %b" id is_localhost;
-           is_localhost, id
+           debug "VM.receive_memory id = %s" id;
```

Comment:
It may be more interesting to log the `final_id` here, since that's the "actual" VM UUID. Or just log both.